src/pipx/util.py

```diff
@@ -174,7 +174,7 @@ def run_subprocess(
     # See https://github.com/pypa/pipx/issues/1164
     # Conversely, if the binary is a symlink, then we should NOT use the real path, as Python expects to receive the
     # symlink in argv[0] so that it can locate the venv.
-    if not os.path.islink(cmd_str_list[0]) and WINDOWS:
```

Comment:
I'm also wondering if we could move this entire logic to `pipx/interpreter:find_python_interpreter`, as to me that seems like the more logical place. 

I've played around with this on my fork, feel free to take a look: https://github.com/Gitznik/pipx/tree/parse_python_to_path